### PR TITLE
#159687856 In app wallet

### PR DESCRIPTION
**Pull request by otseobande**
[//]: # (PR Title should follow this format "#<story-id> story description")

#### What does this PR do?
Allows customers use an in app wallet for transactions
#### Description of Task to be completed?

- add fund wallet feature
- add wallet as a payment option

#### How should this be manually tested?

- Navigate to a user profile page
- Scroll to the wallet section and fund wallet
- Select wallet payment as a payment option from the admin end
- Purchase a product and pay with wallet

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

- #159687856

#### Screenshots or gifs (if appropriate):

screen shot 2018-09-02 at 12 19 58 am

#### Questions:

N/A

### CODE REVIEW

**Review by ozimos — APPROVED** [review]
good work keep it up
However the search box is not displaying on your review app.

**Review by rovilay — COMMENTED** [review]
Good work. Clean code and easy to read. keep it up.
Can you work on fixing the failing test?. Thank you.

**Comment by otseobande** [comment]
Circleci had a downtime loading npm packages last night and that caused the ci to fail. I'll push again and try to get it running.

**Comment by otseobande** [comment]
They are still having the issue 
screen shot 2018-09-02 at 9 39 08 am

File: imports/plugins/custom/in-app-wallet/lib/collections/schemas/accounts.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { SimpleSchema } from "meteor/aldeed:simple-schema";
+import { registerSchema } from "@reactioncommerce/reaction-collections";
+import { accounts } from "/lib/collections/schemas/accounts";
+
+export const Accounts = new SimpleSchema([accounts, {
+  walletBalance: {
```
Comment (ozimos) [inline imports/plugins/custom/in-app-wallet/lib/collections/schemas/accounts.js]: You extend the accounts schema here. Very good. But you also modify the schema directly. See below

**Review by rovilay — APPROVED** [review]

File: lib/collections/schemas/accounts.js [hunk 2]
```diff
@@ -217,6 +217,12 @@ export const Accounts = new SimpleSchema({
       }
     },
     optional: true
+  },
+  walletBalance: {
+    type: Number,
+    decimal: true,
+    defaultValue: 0,
+    optional: true
   }
```
Comment (ozimos) [inline lib/collections/schemas/accounts.js]: You directly modify the accounts schema here. See comments above

File: imports/plugins/custom/in-app-wallet/server/methods.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    if (walletBalance >= amount) {
+      Accounts.update({
+        userId
+      }, {
+        $inc: {
+          walletBalance: (amount * -1)
+        }
+      });
+    } else {
+      Logger.warn("Failed wallet/charge: Insufficient funds. Please fund wallet");
```
Comment (ozimos) [inline imports/plugins/custom/in-app-wallet/server/methods.js:90]: You can use Alerts instead of Logger. reaction Commerce has Alerts

Comment (otseobande) [reply imports/plugins/custom/in-app-wallet/server/methods.js:90]: Nice point but this is server side code. Alerts work on the client.

File: lib/collections/schemas/accounts.js (hunk 2 above)
Comment (otseobande) [reply lib/collections/schemas/accounts.js]: I have removed the extension and stuck with modifying the accounts schema directly. Extending it in my plugin was not working as expected.

File: imports/plugins/custom/in-app-wallet/server/methods.js (hunk 3 above)
Comment (ozimos) [reply imports/plugins/custom/in-app-wallet/server/methods.js:90]: I stand corrected. Carry on

**Pull request merged**